src/corefx/System.Globalization.Native/pal_icushim.c

```diff
@@ -253,41 +254,41 @@ extern "C" int32_t GlobalizationNative_LoadICU()
 #define PER_FUNCTION_BLOCK(fn, lib) \
     static_assert((sizeof(#fn) + MaxICUVersionStringLength + 1) <= sizeof(symbolName), "The symbolName is too small for symbol " #fn); \
     sprintf(symbolName, #fn "%s", symbolVersion); \
-    fn##_ptr = (decltype(fn)*)dlsym(lib, symbolName); \
+    fn##_ptr = (__typeof(fn)*)dlsym(lib, symbolName); \
     if (fn##_ptr == NULL) { fprintf(stderr, "Cannot get symbol %s from " #lib "\nError: %s\n", symbolName, dlerror()); abort(); }
 
     FOR_ALL_ICU_FUNCTIONS
 #undef PER_FUNCTION_BLOCK
 
 #ifdef __APPLE__
-    // libicui18n initialized with libicuuc so we null it to avoid double closing same handle   
-    libicui18n = nullptr;
+    // libicui18n initialized with libicuuc so we null it to avoid double closing same handle
+    libicui18n = NULL;
 #endif // __APPLE__
 
     return 1;
 }
 
 // GlobalizationNative_GetICUVersion
 // return the current loaded ICU version
-extern "C" int32_t GlobalizationNative_GetICUVersion()
+int32_t GlobalizationNative_GetICUVersion()
 {
     int32_t version;
     u_getVersion((uint8_t *) &version);
-    return version; 
+    return version;
 }
 
 __attribute__((destructor))
```

Comment:
@janvorli I think that this shutdown method is broken. There can be code in libicu code running during shutdown. Unloading icu here will cause intermittent crashes during process exit.

Reply:
I have tried to make it crash and there does not appear to be a real problem. `dlclose` during shutdown seems to be ignored - the library is not actually unmapped.